File: ControlStructures/ControlStructures.cs
using System;

class Program
{
      static void Main()
      {
            bool a = true;
            bool b = false;

            if (a)
            {
                  Console.WriteLine("a is true");
            }
            else
            {
                  Console.WriteLine("a is false");
            }

            if (b)
            {
                  Console.WriteLine("b is true");
            }
            else
            {
                  Console.WriteLine("b is false");
            }

            if (a && b)
            {
                  Console.WriteLine("a and b are true");
            }
            else
            {
                  Console.WriteLine("a and b are not both true");
            }

            if (a || b)
            {
                  Console.WriteLine("a or b is true");
            }
            else
            {
                  Console.WriteLine("a and b are both false");
            }

            if (a ^ b)
            {
                  Console.WriteLine("a or b is true, but not both");
            }
            else
            {
                  Console.WriteLine("a and b are both false or both true");
            }

            if (!a)
            {
                  Console.WriteLine("a is false");
            }
            else
            {
                  Console.WriteLine("a is true");
            }

            if (!b)
            {
                  Console.WriteLine("b is false");
            }
            else
            {
                  Console.WriteLine("b is true");
            }

            if (a == b)
            {
                  Console.WriteLine("a and b are equal");
            }
            else
            {
                  Console.WriteLine("a and b are not equal");
            }

            bool c = true;
            bool d = false;

            if (a)
            {
                  if (b)
                  {
                        if (c)
                        {
                              if (d)
                              {
                                    Console.WriteLine("a, b, c, and d are true");
                              }
                              else
                              {
                                    Console.WriteLine("a, b, and c are true, but d is false");
                              }
                        }
                        else
                        {
                              Console.WriteLine("a and b are true, but c is false");
                        }
                  }
                  else
                  {
                        Console.WriteLine("a is true, but b is false");
                  }
            }
            else
            {
                  Console.WriteLine("a is false");
            }


      }
}

File: ControlStructures/ControlStructures.java
public class ControlStructures
{
      public static void main(String[] args)
      {
            boolean a = true;
            boolean b = false;

            if (a)
            {
                  System.out.println("a is true");
            }

            if (b)
            {
                  System.out.println("b is true");
            }

            if (a && b)
            {
                  System.out.println("a and b are true");
            }

            if (a || b)
            {
                  System.out.println("a or b is true");
            }

            if (a && !b)
            {
                  System.out.println("a is true and b is false");
            }

            if (a || !b)
            {
                  System.out.println("a is true or b is false");
            }

            if (a && b)
            {
                  System.out.println("a and b are true");
            }
            else
            {
                  System.out.println("a and b are not both true");
            }

            boolean c = true;
            boolean d = false;

            if (c)
            {
                  System.out.println("c is true");
            }
            else if (d)
            {
                  System.out.println("d is true");
            }
            else
            {
                  System.out.println("c and d are false");
            }

            if (a)
            {
                  if (b)
                  {
                        if (c)
                        {
                              if (d)
                              {
                                    if (a)
                                    {
                                          if (b)
                                          {
                                                if (c)
                                                {
                                                      if (d)
                                                      {
                                                            if (a)
                                                            {
                                                                  if (b)
                                                                  {
                                                                        System.out.println("a and b are true");
                                                                  }
                                                                  else
                                                                  {
                                                                        System.out.println("a is true and b is false");
                                                                  }
                                                            }
                                                            else
                                                            {
                                                                  System.out.println("a is false");
                                                            }
                                                      }
                                                      else
                                                      {
                                                            System.out.println("d is false");
                                                      }
                                                }
                                                else
                                                {
                                                      System.out.println("c is false");
                                                }
                                          }
                                          else
                                          {
                                                System.out.println("b is false");
                                          }
                                    }
                                    else
                                    {
                                          System.out.println("a is false");
                                    }
                              }
                              else
                              {
                                    System.out.println("d is false");
                              }
                        }
                        else
                        {
                              System.out.println("c is false");
                        }
                  }
                  else
                  {
                        System.out.println("b is false");
                  }
            }
            else
            {
                  System.out.println("a is false");
            }

      }
}
